### DISCUSSION TITLE
Multi-arch CGO binaries via self-hosted runners?

### DISCUSSION THREAD

**Discussion by tashian** [goreleaser/goreleaser#2064] [Q&A] [answered]
Hi! We're looking to migrate our project [step-ca](https://github.com/smallstep/certificates) to GoReleaser, and I'm curious about CGO support.

I've seen this done using Docker containers (eg. with `hugo`, see [this issue](https://github.com/goreleaser/goreleaser/issues/708)), and I like that option but it adds a lot of complexity compared to a typical goreleaser config.

But I was kind of curious about an alternative .. I'm wondering if self-hosted GitHub Action runners could be used just for the build step? We could potentially set up build servers for each platform we need to support.

I'm envisioning one goreleaser config for each platform, and those would run in parallel.

But, if we did that, it seems like we'd have to upload the artifacts somewhere at the end, and do the actual release steps with custom code. Which sort of defeats the purpose of using GoReleaser. But, am I missing something here?

**Comment by caarlos0** [comment]
You could do that, the only thing that might be an issue is the checksums file, since each run will generate a different file (because it will build only for a given target).

If you disable the checksums (or generate them with different names on each platform), it should be fine, I think.

**Comment by caarlos0** [comment, marked as answer]
This should now be possible with split/merge pipelines in goreleaser pro: https://carlosbecker.com/posts/goreleaser-split-merge/
